lib/puppetserver/ca/setup_command.rb

```diff
@@ -0,0 +1,136 @@
+require 'optparse'
+require 'puppetserver/ca/x509_loader'
+require 'puppetserver/ca/puppet_config'
+
+module Puppetserver
+  module Ca
+    class SetupCommand
+
+      def initialize(logger)
+        @logger = logger
+      end
+
+      def run!(input)
+        bundle_path = input['cert-bundle']
+        key_path = input['private-key']
+        chain_path = input['crl-chain']
+        config_path = input['config']
+
+        files = [bundle_path, key_path, chain_path, config_path].compact
+
+        errors = validate_file_paths(files)
+        return 1 if log_possible_errors(errors)
+
+        unless chain_path
+          @logger.err 'Warning:'
+          @logger.err '    No CRL chain given'
+          @logger.err '    Full CRL chain checking will not be possible'
```

Comment:
Should we tell them that we will generate a CRL for them based on the cert provided, or is this tmi?  Why wouldnt' full crl chain checking be possible?  It will, right?  But only after some certificates are revoked.

Reply:
The agent differentiates between "leaf" and "full" CRL checking, where with leaf it only checks that the most subordinate CA hasn't been revoked by its issuer and full checking checks that none of the issuers farther up the hierarchy have been revoked by their issuers.

The Puppet Server CA can generate its own CRL, but without being given CRLs for its issuer it can't, and the agents its distributes CRLs to won't, be able to check the full chain's CRLs.

Reply:
Ok.  I was missing the "full" part of the message. Thanks.